### ISSUE TITLE
error code for errors like block number not yet indexed

### ISSUE THREAD

**Issue by wighawag** [graphprotocol/graph-node#1559]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**

Currently when making a query with the block parameter, like 

```
{
  tokens(block: {number: 10000000}) {
    id
    owner
  }
}
```

We get an error like :

```
{
  "errors": [
    {
      "message": "Failed to decode `block.number` value: `subgraph QmX2cspb9qty6nzUEwf7iw6jU7ptwk6hBp9VVhsHdHHcMA has only indexed up to block number 9753342 and data for block number 10000000 is therefore not yet available`"
    }
  ]
}
```

**What is the expected behavior?**

It would be great to have error code so we can easily differentiate between different type of error, instead of parsing the error message

**Comment by wighawag** [comment]
Hi, is there any interest in that.
I just encounter the issue again and was going to submit a new issue when I realised I already submitted the same here :)

**Comment by chanhosuh** [comment]
> Hi, is there any interest in that. I just encounter the issue again and was going to submit a new issue when I realised I already submitted the same here :)

I definitely have interest in this too.  I imagine most people are interested in getting better errors from The Graph.

**Comment by azf20** [comment]
thanks @chanhosuh and @wighawag 
@Jannis @lutter this seems useful, but also relevant to the attestable errors work?

**Comment by lutter** [comment]
Yes, I think machine-readable error codes would be very worthwhile to have
